homeassistant/components/portainer/binary_sensor.py

```diff
@@ -0,0 +1,161 @@
+    PortainerBinarySensorEntityDescription(
+        key="status",
+        translation_key="status",
+        state_fn=lambda data: data.state == "running",
+        device_class=BinarySensorDeviceClass.RUNNING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+ENDPOINT_SENSORS: tuple[PortainerBinarySensorEntityDescription, ...] = (
+    PortainerBinarySensorEntityDescription(
+        key="status",
+        translation_key="status",
+        state_fn=lambda data: data.endpoint.status == 1,  # 1 = Running | 2 = Stopped
+        device_class=BinarySensorDeviceClass.RUNNING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: PortainerConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Portainer binary sensors."""
+    coordinator = entry.runtime_data
+    entities: list[BinarySensorEntity] = []
+
+    for endpoint in coordinator.data.values():
+        entities.extend(
+            PortainerEndpointSensor(
+                coordinator=coordinator,
+                entity_description=entity_description,
+                device_info=endpoint,
+            )
+            for entity_description in ENDPOINT_SENSORS
+        )
+
+        assert endpoint.containers
+        entities.extend(
+            [
+                PortainerContainerSensor(
+                    coordinator=coordinator,
+                    entity_description=entity_description,
+                    device_info=container,
+                    via_device=endpoint,
+                )
+                for container in endpoint.containers.values()
+                for entity_description in CONTAINER_SENSORS
+            ]
+        )
+
+    async_add_entities(entities)
+
+
+class PortainerEndpointSensor(PortainerEndpointEntity, BinarySensorEntity):
+    """Representation of a Portainer endpoint binary sensor entity."""
+
+    entity_description: PortainerBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PortainerCoordinator,
+        entity_description: PortainerBinarySensorEntityDescription,
+        device_info: PortainerCoordinatorData,
+    ) -> None:
+        """Initialize Portainer endpoint binary sensor entity."""
+        self.entity_description = entity_description
+        super().__init__(device_info, coordinator)
+
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{device_info.id}_{entity_description.key}"
+
+    @property
+    def available(self) -> bool:
+        """Return if the device is available."""
+        return super().available and self.device_id in self.coordinator.data
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+
+        return (
+            self.entity_description.state_fn(device_info)
+            if (device_info := self.coordinator.data[self.device_id])
+            else None
+        )
```

Comment:
In theory device_info would now be there right? So we don't need the if statement anymore right?